src/mcpadapt/smolagents_adapter.py

```diff
@@ -77,24 +179,55 @@ def adapt(
             A SmolAgents tool.
         """
 
+        # make sure jsonref are resolved
+        input_schema = {
+            k: v
+            for k, v in jsonref.replace_refs(mcp_tool.inputSchema).items()
+            if k != "$defs"
+        }
+
+        # make sure mandatory `description` and `type` is provided for each arguments:
+        for k, v in input_schema["properties"].items():
+            if "description" not in v:
+                input_schema["properties"][k]["description"] = "see tool description"
+            if "type" not in v:
+                input_schema["properties"][k]["type"] = "string"
+
+        # Extract and resolve outputSchema if present (only when structured_output=True)
+        output_schema = None
+        if self.structured_output and hasattr(mcp_tool, 'outputSchema') and mcp_tool.outputSchema:
+            try:
+                output_schema = jsonref.replace_refs(mcp_tool.outputSchema)
+            except Exception as e:
+                logger.warning(f"Failed to resolve outputSchema for tool {mcp_tool.name}: {e}")
+                output_schema = mcp_tool.outputSchema  # Use unresolved schema as fallback
+
+        # Determine output_type based on mode and schema
+        if self.structured_output and output_schema:
+            output_type = "object"  # Structured data expected
+        else:
+            output_type = "string"  # Plain text expected
```

Comment:
I don't see any reason to make this check as the `output_type` should be `object` so that `ImageContent` and `AudioContent` work.

Reply:
Based on this line in the specification 
```
For backwards compatibility, a tool that returns structured content SHOULD also return the serialized JSON in a TextContent block.
```

A structured content should only exist on TextContent.
So we can just check for that `structureContent` in the `TextContent`, as a JSON result value should be serializable as a string anyways.